google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/scaladsl/GooglePubSub.scala

```diff
@@ -27,61 +28,46 @@ protected[pubsub] trait GooglePubSub {
   def publish(topic: String, config: PubSubConfig, parallelism: Int = 1)(
       implicit actorSystem: ActorSystem,
       materializer: Materializer
-  ): Flow[PublishRequest, immutable.Seq[String], NotUsed] = {
-    import materializer.executionContext
-
-    if (httpApi.isEmulated) {
-      Flow[PublishRequest].mapAsyncUnordered(parallelism) { request =>
-        httpApi.publish(config.projectId, topic, maybeAccessToken = None, request)
-      }
-    } else {
-      Flow[PublishRequest].mapAsyncUnordered(parallelism) { request =>
-        config.session.getToken().flatMap { accessToken =>
-          httpApi.publish(config.projectId, topic, Some(accessToken), request)
-        }
+  ): Flow[PublishRequest, immutable.Seq[String], NotUsed] =
+    httpApi
+      .accessToken[PublishRequest](config, parallelism)
+      .via(httpApi.publish(config.projectId, topic, parallelism))
+      .mapAsyncUnordered(parallelism) {
```

Comment:
This should be a `mapAsync(1)` to be sure the response order is kept.

Reply:
I'll add context than this should stay as it is.

Reply:
I make leave it only for `publish` and change to `mapAsync(1)` for others